src/main/java/com/minecolonies/colony/Field.java

```diff
@@ -0,0 +1,467 @@
+                                            8+j*18, 84+i*18));
+            }
+        }
+
+        for (i = 0; i < 9; ++i)
+        {
+            addSlotToContainer(new Slot(playerInventory, i, 8 + i * 18,
+                                        142));
+        }
+        calculateSize(world,location.down());
+    }
+
+    /**
+     * Calculates recursively the length of the field until a certain point.
+     * @param position the start position.
+     * @param world the world the field is in.
+     */
+    public void calculateSize(World world, BlockPos position)
+    {
+        //Calculate in all 4 directions
+        this.lengthPlusX = searchNextBlock(0, position.east(), EnumFacing.EAST, world);
+        this.lengthMinusX= searchNextBlock(0, position.west(), EnumFacing.WEST, world);
+        this.widthPlusZ  = searchNextBlock(0, position.south(), EnumFacing.SOUTH, world);
+        this.widthMinusZ = searchNextBlock(0, position.north(), EnumFacing.NORTH, world);
+    }
+
+    /**
+     * Calculates the field size into a specific direction.
+     * @param blocksChecked how many blocks have been checked.
+     * @param position the start position.
+     * @param direction the direction to search.
+     * @param world the world object.
+     * @return the distance.
+     */
+    private int searchNextBlock(int blocksChecked, BlockPos position, EnumFacing direction, World world)
+    {
+        if(blocksChecked == getMaxRange() || isNoPartOfField(world, position))
+        {
+            return blocksChecked;
+        }
+        return  searchNextBlock(blocksChecked+1,position.offset(direction),direction,world);
+    }
+
+    /**
+     * Checks if a certain position is part of the field. Complies with the definition of field block.
+     * @param world the world object.
+     * @param position the position.
+     * @return true if it is.
+     */
+    public boolean isNoPartOfField(World world, BlockPos position)
+    {
+        return world.isAirBlock(position) || world.getBlockState(position.up()).getBlock().getMaterial().isSolid();
+    }
+
+    /**
+     * Returns the {@link BlockPos} of the current object, also used as ID
+     *
+     * @return          {@link BlockPos} of the current object
+     */
+    public BlockPos getID()
+    {
+        // Location doubles as ID
+        return this.location;
+    }
+
+    /**
+     * Returns the colony of the field
+     *
+     * @return          {@link com.minecolonies.colony.Colony} of the current object
+     */
+    public Colony getColony()
+    {
+        return this.colony;
+    }
+
+    /**
+     * Create and load a Field given it's saved NBTTagCompound
+     *
+     * @param colony    The owning colony
+     * @param compound  The saved data
+     * @return          {@link com.minecolonies.colony.Field} created from the compound.
+     */
+    public static Field createFromNBT(Colony colony, NBTTagCompound compound)
+    {
+        Field field = new Field(colony);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'field' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)